packages/lesmis/src/api/mutations/useLogin.js

```diff
@@ -21,9 +23,13 @@ export const useLogin = () => {
       }),
     {
       onSuccess: () => {
-        queryClient.resetQueries('user');
-        queryClient.resetQueries('entity');
-        queryClient.resetQueries('entities');
+        queryClient.clear();
+
+        if (history.location?.state?.referer) {
+          history.push(history.location.state.referer);
```

Comment:
Nice! Might be worth a comment above for people walking in without context, though it's pretty clean and clear